pallets/msa/src/offchain_storage.rs

```diff
@@ -0,0 +1,135 @@
+/// Offchain Storage for MSA
+use codec::{Decode, Encode};
+use common_primitives::offchain as offchain_common;
+use sp_runtime::offchain::storage::StorageRetrievalError;
+use sp_std::{
+	collections::btree_map::BTreeMap,
+	fmt::{Debug, Formatter},
+	vec::Vec,
+};
+
+/// Generic prefix for MSA index storage
+pub const MSA_INDEX_KEY: &[u8] = b"frequency::msa::";
+
```

Comment:
this could be changed for unique storage keys based on storage types of K,V, check the usage below